### PR TITLE
Add request start time to CoyoteRequest

### LINKED ISSUE #4182: Incorrect AccessLogValve output with Armeria Tomcat connector
I'm trying to use Armeria Tomcat connector (armeria-tomcat9) with Spring Boot.
Tomcat's Access Log Valve (https://tomcat.apache.org/tomcat-9.0-doc/config/valve.html#Access_Log_Valve) is used to output access log on our server.
However, output value of `%D` (processing time) is broken when I use it with armeria-tomcat9. It seems the value is timestamp rather than processing time.

```
pattern: timestamp:%{msec}t elapsed_time:%D
output:  timestamp:1648839887052 elapsed_time:1648839887053
```

We set `AccessLogValve` like this:

```
@Configuration
public class WebMvcConfiguration implements WebServerFactoryCustomizer<TomcatServletWebServerFactory> {
    @Override
    public void customize(TomcatServletWebServerFactory factory) {
        AccessLogValve valve = new AccessLogValve();
        valve.setPattern(settings.getPattern());
        valve.setDirectory(settings.getDirectory());
        ...
        factory.addEngineValves(valve);
    }
}
```

Using Armeria's access log writer is a solution of this problem. It works well with armeria-tomcat9.

### PR DESCRIPTION
**Pull request by kojilin**
Motivation:

As #4182 , Tomcat AccessLog need to use CoyoteRequest's startTime

Modifications:

- Set startTime of CoyoteRequest

Result:

- Closes #4182

### CODE REVIEW

**Comment by kojilin** [comment]
Any suggestion for unit test? I just test in example project by adding setting like issue link.

**Comment by trustin** [comment]
> Any suggestion for unit test? I just test in example project by adding setting like issue link.

I have no good idea, either. How about adding a `.jsp` file that prints the value of `request.getCoyoteRequest().getStartTime()` and adding a test that makes sure it's non-zero?

File: tomcat9/src/main/java/com/linecorp/armeria/server/tomcat/TomcatService.java [hunk 1]
```diff
@@ -516,6 +517,9 @@ private Request convertRequest(ServiceRequestContext ctx, AggregatedHttpRequest
         final HttpData content = req.content();
         coyoteReq.setInputBuffer((InputBuffer) INPUT_BUFFER_CONSTRUCTOR.invoke(content));
 
+        // Set the start time which is used by Tomcat access logging
+        coyoteReq.setStartTime(ctx.log().ensureAvailable(RequestLogProperty.REQUEST_START_TIME)
+                                  .requestStartTimeMillis());
```
Comment (ikhoon) [inline tomcat9/src/main/java/com/linecorp/armeria/server/tomcat/TomcatService.java:522]: How about setting the request start time before returning a null value so that the invalid request is logged with the start time?

File: tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceAccessLogTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        accessLogValve.reset();
+    }
+
+    @Test
+    void haveCorrectProcessingTime() throws Exception {
+        try (CloseableHttpClient hc = HttpClients.createMinimal()) {
```
Comment (ikhoon) [inline tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceAccessLogTest.java:107]: nit: Why not use Armeria's `WebClient`?

**Review by ikhoon — APPROVED** [review]
Thanks, @kojilin!

File: tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceAccessLogTest.java (hunk 2 above)
Comment (kojilin) [reply tomcat9/src/test/java/com/linecorp/armeria/server/tomcat/TomcatServiceAccessLogTest.java]: ah due to copy from existing UnmanagedTomcatServiceTest.
Update to armeria one^^

**Review by jrhee17 — APPROVED** [review]
Thanks @kojilin ! 🙇 👍 🙇

**Review by trustin — APPROVED** [review]
:+1:

**Comment by trustin** [comment]
Thanks, @kojilin 🙇
